src/type-hierarchy.ts

```diff
@@ -239,15 +302,27 @@ class TypeHierarchyProvider implements
       direction: TypeHierarchyDirection.Both
     });
     if (item) {
-      this.root = item;
+      this.startingItem = item;
+      this.root = this.computeRoot();
+
       this._onDidChangeTreeData.fire(null);
 
       // This focuses the "explorer" view container which contains the
       // type hierarchy view.
       vscode.commands.executeCommand('workbench.view.explorer');
 
       // This expands and focuses the type hierarchy view.
-      this.treeView.reveal(this.root, {focus: true});
+      // Focus the item on which the operation was invoked, not the
+      // root (which could be its ancestor or the dummy node).
+      this.treeView.reveal(this.startingItem, {focus: true})
+          .then(() => {}, (reason) => {
+            // Sometimes TreeView.reveal() fails. It's unclear why, and it does
+            // not appear to have any visible effects, but vscode complains if
+            // you don't handle the rejection promise, so we do so and log a
+            // warning.
```

Comment:
this is very unfortunate, but it seems there are no visible effects, we can live with that. 

btw, I see some logs (not sure they are helpful)

```
Warning: TreeView.reveal() failed for reason: Error: Cannot resolve tree item for element 0/0:RecursiveASTVisitor<(anonymous namespace)::UnbridgedCastRewriter>/0:UnbridgedCastRewriter
type-hierarchy.js:250
Warning: TreeView.reveal() failed for reason: Error: TreeError [clangd.typeHierarchyView] Data tree node not found: [object Object]
```